mortar/src/main/java/mortar/MortarScope.java

```diff
@@ -16,49 +16,48 @@
 package mortar;
 
 import android.content.Context;
+import java.util.LinkedHashMap;
+import java.util.Map;
 
-public interface MortarScope {
-  String ROOT_NAME = "Root";
+import static java.lang.String.format;
 
-  /** Returns the name of this scope. */
-  String getName();
+public interface MortarScope {
+  final String ROOT_NAME = "Root";
+  final String DIVIDER = ":";
 
   /**
-   * Returns the graph for this scope.
-   *
-   * @throws IllegalStateException if this scope has been destroyed
+   * Returns the name of this scope, used to retrieve it from its parent via {@link
+   * #findChild(String)}.
    */
-  <T> T getObjectGraph();
+  String getName();
+
+  String getPath();
+
+  <T> T getService(String serviceName);
 
   /**
-   * Register the given {@link Scoped} instance to have its {@link Scoped#onExitScope()}
-   * method called from {@link MortarScope#destroyChild(MortarScope)}. Redundant calls are safe,
-   * they will not lead to double registration.
+   * Register the given {@link Scoped} instance to have its {@link Scoped#onEnterScope(MortarScope)}
+   * and {@link Scoped#onExitScope()} methods called. Redundant registrations are safe,
+   * they will not lead to additional calls to these two methods.
+   * <p>
+   * Calls to {@link Scoped#onEnterScope(MortarScope) onEnterScope} are dispatched asynchronously
+   * if a previous {@code register} call is already in progress.
    *
    * @throws IllegalStateException if this scope has been destroyed
    */
   void register(Scoped scoped);
 
   /**
-   * Returns the child instance whose name matches the given, or null if there is none.
+   * Returns the child scope whose name matches the given, or null if there is none.
    *
    * @throws IllegalStateException if this scope has been destroyed
    */
   MortarScope findChild(String name);
 
-  /**
-   * Creates a new child scope based on the given name and subgraph.
-   *
-   * @throws IllegalStateException if this scope has been destroyed
-   */
-  MortarScope createChild(String childName, Object childObjectGraph);
+  Builder buildChild(String name);
```

Comment:
Yeah, this here will prevent reuse of the builder.  Move the `name` parameter to `Builder#build()` and you can make as many similarly-but-slightly-differently-configured children as you want with the same builder instance.